### PR TITLE
Equality for tagless and update Quick Start

### LINKED ISSUE #476: Update README to make @tagless and @free support equally important.

### LINKED ISSUE #486: Update Quick Start embedding the program in the module
https://gitter.im/47deg/freestyle?at=5a42d6d90163b02810802339

### PR DESCRIPTION
**Pull request by AdrianRaFo**
Fixes #476 
Fixes #486

### CODE REVIEW

File: docs/src/main/tut/docs/README.md [hunk 1]
```diff
@@ -63,10 +63,30 @@ Learn more about [algebras](./core/algebras) in the extended documentation.
...
+
+The example below combines both algebras to produce a more complex program:
+
 ```tut:book
 @module trait Application {
   val validation: Validation
   val interaction: Interaction
+  
+  def program: FS.Seq[Unit] = {
+  import cats.implicits._
```
Comment (raulraja) [inline docs/src/main/tut/docs/README.md]: improperly formatted

File: docs/src/main/tut/docs/core/handlers/README.md [hunk 2]
```diff
@@ -139,6 +139,139 @@ program[Backend.Op].interpret[KVStoreState]
...
+}
+```
+
+Once all of our algebras are considered, we can execute our programs
+
+```tut:book
+taglessProgram[App.Op].interpret[Try]
+```
+
 ## A note on performance
```
Comment (raulraja) [inline docs/src/main/tut/docs/core/handlers/README.md]: We can remove this as our optimizations take care of most of it.

File: docs/src/main/tut/docs/core/modules/README.md [hunk 3]
```diff
@@ -40,8 +45,8 @@ import algebras._
 
 object modules {
     @module trait Persistence {
-      val database: Database
-      val cache: Cache
+      val database: Database.StackSafe
```
Comment (raulraja) [inline docs/src/main/tut/docs/core/modules/README.md:48]: There is modules for tagless and therefore this isn't needed anymore to use the StackSafe representation to embed them into modules

File: docs/src/main/tut/docs/core/modules/README.md [hunk 4]
```diff
@@ -108,7 +116,7 @@ If you were to create this by hand, in the case of the example above, it will lo
 import iota._
 import TListK.:::
 type ManualAppCoproduct[A] = CopK[
-  IdValidation.Op ::: Presenter.Op ::: Cache.Op ::: Database.Op ::: TNilK, A]
+  IdValidation.Op ::: Presenter.Op ::: Cache.StackSafe.Op ::: Database.StackSafe.Op ::: TNilK, A]
```
Comment (raulraja) [inline docs/src/main/tut/docs/core/modules/README.md]: There is no need to use Coproduct with tagless algebras.

**Review by raulraja — CHANGES_REQUESTED** [review]

File: docs/src/main/tut/docs/README.md [hunk 5]
```diff
@@ -126,7 +123,7 @@ import cats.implicits._
 import scala.concurrent.duration.Duration
 import scala.concurrent.Await
 
-val futureValue = program[Application.Op].interpret[Future]
+val futureValue = Application.instance.program.interpret[Future]
```
Comment (raulraja) [inline docs/src/main/tut/docs/README.md]: Can we change this to `cats.effect.IO`?

**Review by raulraja — APPROVED** [review]

**Comment by AdrianRaFo** [comment]
@MaureenElsberry Can you check the docs, please?

File: docs/src/main/tut/docs/README.md [hunk 6]
```diff
@@ -76,44 +95,21 @@ Once you have these abstract definitions, you can combine them in whichever way
...
-            interaction.tell("awesomesauce!")
-         else
-            interaction.tell(s"$userInput is not valid")
-  } yield ()
-}
-```
-
 ## Running programs
 
 In order to run programs, we need interpreters. We define interpreters providing implementations for the operations defined in our algebras:
```
Comment (MaureenElsberry) [inline docs/src/main/tut/docs/README.md]: one question here... is this supposed to be saying that we're defining interpreters that are providing implementations .... or,      we define interpreters by providing .... ??

Comment (AdrianRaFo) [reply docs/src/main/tut/docs/README.md]: The second, we define interpreter and inside we provide the implementations

File: docs/src/main/tut/docs/README.md [hunk 7]
```diff
@@ -123,16 +119,15 @@ At this point, we can run our pure programs at the edge of the world:
 
 ```tut:book
 import cats.implicits._
-import scala.concurrent.duration.Duration
-import scala.concurrent.Await
+import cats.effect.IO
+import cats.effect.IO._
```
Comment (peterneyens) [inline docs/src/main/tut/docs/README.md]: I don't think we need the last import?

File: docs/src/main/tut/docs/README.md [hunk 8]
```diff
@@ -63,10 +63,29 @@ Learn more about [algebras](./core/algebras) in the extended documentation.
...
 @module trait Application {
   val validation: Validation
   val interaction: Interaction
+  
+  import cats.implicits._
+  
+  def program: FS.Seq[Unit] = 
+    for {
+      userInput <- interaction.ask("Give me something with at least 3 chars and a number on it")
+      valid     <- (validation.minSize(userInput, 3) |@| validation.hasNumber(userInput)).map(_ && _).freeS
```
Comment (peterneyens) [inline docs/src/main/tut/docs/README.md]: We should replace the cartesian builder syntax (the `|@|`) here (like you did in the _Combining `@tagless` and `@free` algebras_ section).

```scala
(validation.minSize(userInput, 3), validation.hasNumber(userInput)).mapN(_ && _).freeS
```

We could also do (but I think the for comprehension is better in this documentation) : 

```scala
interaction.ask("Give me something with at least 3 chars and a number on it").flatMap { userInput =>
  (validation.minSize(userInput, 3), validation.hasNumber(userInput))
    .mapN(_ && _)
    .freeS
    .ifM(
      interaction.tell("awesomesauce!"),
      interaction.tell(s"$userInput is not valid")
    )
}
```

**Comment by AdrianRaFo** [comment]
@peterneyens Everything LGTM now?

**Review by peterneyens — APPROVED** [review]

**Pull request merged**